### PR TITLE
Vivoapi

### PR DESCRIPTION
**Pull request by delsborg**
This pull request removes the Fuseki SPARQL endpoint dependency.
It now queries the VIVO sparqlQueryAPI endpoint with a username/password.
This is not only more secure, but also eliminates risk in case the Fuseki endpoint, a separate process, is down.
This change also:
Introduces a configuration file, vivoapipw.py, which holds various secrets and instance specific configuration data. The file itself isn't checked in but a template vivoapipw.py.example file is provided.
The python people and publication ingest scripts have all instance specific dependencies removed and they are now passed as parameters either from the calling script or from importing the vivoapipw.py file.

### CODE REVIEW

File: ingest/ingest-publications.py [hunk 1]
```diff
@@ -320,34 +334,36 @@ def process_publication(publication, endpoint='http://localhost:2020/ds/sparql')
...
-g1 = g1 + describe(endpoint,get_subjects_query)
-g1 = g1 + describe(endpoint,get_author_query)
-g1 = g1 + describe(endpoint,get_pub_query)
-
 if __name__ == "__main__":
     parser = argparse.ArgumentParser()
-    parser.add_argument('--threads', default=12, help='number of threads to use (default = 6)')
-    parser.add_argument('--sparql', default='http://localhost:2020/ds/sparql', help='sparql endpoint')
+    parser.add_argument('--threads', default=16, help='number of threads to use (default = 12)')
+    parser.add_argument('--sparqlendpoint', default='http://yourhost:8780/vivo/api/sparqlQuery', help='sparql endpoint')
```
Comment (aviggio) [inline ingest/ingest-publications.py]: Is the "yourhost:8780" an example or hardcoded value? Maybe add a comment inline here to call that out?

Comment (delsborg) [reply ingest/ingest-publications.py]: this is an example because the code is public, the actual URL is in the secrets file

Comment (delsborg) [reply ingest/ingest-publications.py]: added more info in the help for the --sparql parameter

File: ingest/vivoapipw.py.example [hunk 2]
```diff
@@ -0,0 +1,8 @@
+#Entries to access to vivo sparql update api
+# Copy this to vivoapipw.py and modify the entries
```
Comment (aviggio) [inline ingest/vivoapipw.py.example]: Could add a comment here noting that this file is used by both shell and Python scripts

**Pull request merged**